### DISCUSSION TITLE
Only one of two rules executed on System#Init

### DISCUSSION THREAD

**Discussion by shaman79** [arendst/Tasmota#22102] [Support request] [answered]
Hi there,
I have built Tasmota for thermostat using the recommended overrides as follows:

```
#define USER_RULE1 "ON System#Init DO ThermostatModeSet 1 ENDON"
#define USER_RULE2 "ON System#Init DO SensorInputSet 1 ENDON"

#define USER_BACKLOG "Rule1 1; Rule1 4; Rule2 1; Rule2 4"
```

The problem is that only first rule is getting executed as per serial log:

```
12:23:10]00:00:00.619 RSL: POWER = OFF
[12:23:10]00:00:00.743 RUL: SYSTEM#INIT performs 'ThermostatModeSet 1'
[12:23:10]00:00:00.751 RSL: RESULT = {"ThermostatModeSet1":1}
[12:23:10]00:00:00.040 WIF: WifiManager active for 3 minutes
```

What am I missing here?

**Comment by lalo-uy** [comment]
Why not:
#define USER_RULE1 "ON System#Init DO Backlog: ThermostatModeSet 1  ;
SensorInputSet
1 ENDON"

**Reply by shaman79** [reply]
That could be a workaround ;) I would like to know why it is not working the standard way tho.

**Reply by sfromis** [reply]
Still, do avoid the syntax error, there is no `:` in the syntax.

**Reply by shaman79** [reply]
Funny enough, with the proposed workaround only the second command got executed.

```
[11:00:58]00:00:00.616 RSL: POWER = OFF
[11:00:58]00:00:00.740 RUL: SYSTEM#INIT performs 'Backlog "ThermostatModeSet 1;SensorInputSet 1"'
[11:00:58]00:00:00.836 RSL: RESULT = {"SensorInputSet1":1}
```

So, either I have some syntax error that I could not see or there is a bug. 
Can someone help me with this?

Relevant part of the tasmota/user_config_override.h is below.

```
#define USER_RULE1 "ON System#Init DO Backlog \"ThermostatModeSet 1;SensorInputSet 1\" ENDON"
#define USER_BACKLOG "Rule1 1; Rule1 4"
```

**Reply by shaman79** [reply, marked as answer]
It was a bit of trial snd error, but this works:

```
#define USER_RULE1 "ON System#Init DO Backlog ThermostatModeSet 1; SensorInputSet 1 ENDON"
#define USER_BACKLOG "Rule1 1; Rule1 4" 
```
